### PR TITLE
feat: Add descriptions to all system defined GQL stuff

### LINKED ISSUE #796: Add descriptions to all gql types
GQL types have a description field, and it is left empty for almost all of ours.  We should populate it as it shows up in some GQL clients.

### PR DESCRIPTION
**Pull request by AndrewSisley**
## Relevant issue(s)

Resolves #796

## Description

Adds descriptions to all system defined GQL stuff (fields, args, enum values, etc). Also adds a couple of missing tests that I needed in order to be sure of the behaviour I was documenting.

I'm not terribly fussed about the location of the constant declarations, this will need to be redone once Pavneet and co have decided on how they wish to manage these long-term.

All the descriptions have been added in the last commit only, other commits include misc changes (such as tests).

Tested via Altair:

image

### CODE REVIEW

File: request/graphql/schema/types/descriptions.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	notNullStringOperatorBlockDescription string = `
+These are the set of filter operators available for use when filtering on String!
+ values.
+`
+	idOperatorBlockDescription string = `
+These are the set of filter operators available for use when filtering on ID
+ values.
+`
+	eqOperatorDescription string = `
+The equality operator, if the target matches the value the check will pass.
```
Comment (fredcarle) [inline request/graphql/schema/types/descriptions.go:161]: todo: From here to line 205, I would change `, if` to `. If`.

File: request/graphql/schema/types/descriptions.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+Sort the results in ascending order, e.g. null,1,2,3,a,b,c.
+`
+	descOrderDescription string = `
+Sort the results in descending order, e.g. c,b,a,3,2,1,null.
+`
+	primaryDirectiveDescription string = `
+Indicate the primary side of a one-to-one relationship.
+`
+	relationDirectiveDescription string = `
+Allows the explicit definition of relationship attributes, instead of using the system generated
```
Comment (fredcarle) [inline request/graphql/schema/types/descriptions.go:217]: todo: remove comma.

**Review by fredcarle — CHANGES_REQUESTED** [review]
Just a few todos related to the text in the descriptions.

Thanks for adding the tests related to empty sets.

File: request/graphql/schema/descriptions.go [hunk 3]
```diff
@@ -74,3 +74,111 @@ var (
...
+
+const (
+	dockeyArgDescription string = `
+An optional dockey parameter for this field. Only documents with
+ the given dockey will be returned.  If no documents match, the result
+ will be null/empty.
+`
+	dockeysArgDescription string = `
+An optional set of dockeys for this field. Only documents with a dockey
+ matching an dockey in the given set will be returned.  If no documents match,
```
[the remark this answers is not in the document]
Comment (AndrewSisley) [reply request/graphql/schema/descriptions.go:86]: - [ ] a dockey

File: request/graphql/schema/descriptions.go [hunk 4]
```diff
@@ -74,3 +74,111 @@ var (
...
+ not found then an empty set will be returned.
+`
+	singleFieldFilterArgDescription string = `
+An optional filter for this join, if the related record does
+ not meet the filter criteria the host record will still be returned,
+ but the value of this field will be null.
+`
+	listFieldFilterArgDescription string = `
+An optional filter for this join, if none of the related records do
+ not meet the filter criteria the host record will still be returned,
```
[the remark this answers is not in the document]
Comment (AndrewSisley) [reply request/graphql/schema/descriptions.go:104]: - [ ] rm extra words

File: request/graphql/schema/types/descriptions.go [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+The Name of the field that this linked commit mutated.
+`
+	commitLinkCIDFieldDescription string = `
+The CID of this linked commit.
+`
+	commitFieldsEnumDescription string = `
+These are the set of fields supported for grouping by in a commits query.
+`
+	commitsQueryDescription string = `
+Returns a set of commits matching any provided criteria, if no arguments are
```
[the remark this answers is not in the document]
Comment (AndrewSisley) [reply request/graphql/schema/types/descriptions.go:97]: :) feels a bit like a nitpick lol, but will change :grin: 

- [ ] `.`

File: request/graphql/schema/types/descriptions.go [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+ provided all commits in the system will be returned.
+`
+	latestCommitsQueryDescription string = `
+Returns a set of head commits matching any provided criteria, if no arguments are
```
[the remark this answers is not in the document]
Comment (AndrewSisley) [reply request/graphql/schema/types/descriptions.go:101]: - [ ] `.`

File: request/graphql/schema/types/descriptions.go (hunk 2 above)
Comment (AndrewSisley) [reply request/graphql/schema/types/descriptions.go:217]: - [ ] rm comma

File: request/graphql/schema/types/descriptions.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (fredcarle) [reply request/graphql/schema/types/descriptions.go:97]: It just that it feels like 2 sentences so the comma feels like the wrong break.

File: request/graphql/schema/types/descriptions.go (hunk 1 above)
Comment (fredcarle) [reply request/graphql/schema/types/descriptions.go:161]: It feels like it should be all part of the same sentence. "-" is totally fine too.

File: request/graphql/schema/types/descriptions.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (fredcarle) [reply request/graphql/schema/types/descriptions.go:97]: I think that I might be a bit more picky here because it's publicly displayed text 😅

File: request/graphql/schema/types/descriptions.go (hunk 1 above)
Comment (AndrewSisley) [reply request/graphql/schema/types/descriptions.go:161]: changed

**Review by shahzadlone — APPROVED** [review]
Cheers for taking care of these.

**Review by fredcarle — APPROVED** [review]
Thanks for doing the changes Andy!
